src/components/DataViewer/DataViewer.tsx

```diff
@@ -0,0 +1,104 @@
+"use client";
+
+import React from "react";
+import styled from "styled-components";
+import Modal from "react-modal";
+
+interface DataViewerProps {
+    data: { [key: string]: string | number | null };
```

Comment:
We have added comment as suggested

Reply:
I don't think this is necessarily needed - I wouldn't expect a nested JSON to be rendered in a single-level table at all, and the type errors 'are' the sign that it won't work! I think specifying that data is a key-value (where value is string/num/null) should be sufficient.

(Especially since we are not exporting this component or making others use it, there won't be 'downstream' users using it for weird edge cases - the component is made in a way which is intuitive and easy-to-use, and it's on the caller to format their data to fit what the table is expecting!)

Reply:
Ok. We have removed the comment then